lib/grape/api.rb

```diff
@@ -111,8 +106,21 @@ def method_missing(method, *args, &block)
         end
       end
 
+      def compile!
+        require 'grape/eager_load'
+        instance_for_rack.compile! # See API::Instance.compile!
+      end
+
       private
 
+      def instance_for_rack
+        if never_mounted?
+          base_instance
+        else
+          mounted_instances.first
```

Comment:
Should this be all instances and the method to return an array?

Reply:
hmm. I can't remember why this should ever return one of the mounted instances, as you probably want to always be interacting with rack on a predictable instance. I'll have to see what happens if I have this always be the `base_instance`

Reply:
OK so this change was initially introduced in: https://github.com/ruby-grape/grape/pull/1893/files 
The reason why this is needed is that we used to support - before remounting was introduced -  the ability to pre-mount the API  you will run in a namespace (and I was maintaing backwards compatibility).

Basically you are allowed to do
```ruby
class ToBeMountedinRack < Grape::API
  get 'endpoint'
end

class Namespace < Grape::API
  namespace 'namespace' do 
    mount ToBeMountedinRack
   end
end

run ToBeMountedinRack
```

And the expectation is that you will be able to get `/namespace/endpoint` 
I am not sure I want to maintain this behaviour forever, as it doesn't really play nicely with re-mounting an endpoint and I don't find it very sensible (I would find it sensible to `run Namespace` if you want the above behavour, rather than mounting `ToBeMountedinRack`)

Nonetheless, I don't think this is the appropiate PR to introduce this change

Reply:
Agreed. We can deprecate this separately.